File: frontend/src/services/authService.js
import api from './api';

export const login = async (credentials) => (await api.post('/api/auth/login', credentials)).data;
export const register = async (request) => (await api.post('/api/auth/register', request)).data;
export const verifyEmail = async (token) => (await api.get('/api/auth/verify-email', { params: { token } })).data;

File: src/main/java/com/dayflow/hrms/service/AuthService.java
package com.dayflow.hrms.service;

import com.dayflow.hrms.dto.auth.AuthResponse;
import com.dayflow.hrms.dto.auth.LoginRequest;
import com.dayflow.hrms.dto.auth.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    String verifyEmail(String token);
}
